asm/f64/asm_test.go

```diff
@@ -65,3 +74,119 @@ func nonStridedWrite(x []float64, inc int) bool {
 	}
 	return false
 }
+
+// guardVector copies the source vector (vec) into a new slice with guards.
+// Guards guarded[:gdLn] and guarded[len-gdLn:] will be filled with sigil value gdVal.
+func guardVector(vec []float64, gdVal float64, gdLn int) (guarded []float64) {
+	guarded = make([]float64, len(vec)+gdLn*2)
+	copy(guarded[gdLn:], vec)
+	for i := 0; i < gdLn; i++ {
+		guarded[i] = gdVal
+		guarded[len(guarded)-1-i] = gdVal
+	}
+	return guarded
+}
+
+// isValidGuard will test for violated guards, generated by guardVector.
+func isValidGuard(vec []float64, gdVal float64, gdLn int) bool {
+	for i := 0; i < gdLn; i++ {
+		if !same(vec[i], gdVal) || vec[len(vec)-1-i] != gdVal {
+			return false
+		}
+	}
+	return true
+}
+
+// guardIncVector copies the source vector (vec) into a new incremented slice with guards.
+// End guards will be length gdLen.
+// Internal and end guards will be filled with sigil value gdVal.
+func guardIncVector(vec []float64, gdVal float64, inc, gdLen int) (guarded []float64) {
+	if inc < 0 {
+		inc = -inc
+	}
+	inrLen := len(vec) * inc
```

Comment:
Yes, and there are only `len(vec)-1` in-betweens, so it should be enough to have `inrLen := (len(vec)-1)*inc + 1` with care taken when `len(vec) == 0`. However, for testing it doesn't matter if the guard value comes from in-between or from an end-guard, so it's probably ok to leave as is. I'm just noting this to ask if you are aware of this.